hazelcast/src/main/java/com/hazelcast/jet/impl/JobExecutionService.java

```diff
@@ -277,28 +278,23 @@ public CompletableFuture<RawJobMetrics> runLightJob(
                     x -> new ExecutionContext(nodeEngine, jobId, executionId, true));
         }
 
-        try {
-            Set<Address> addresses = participants.stream().map(MemberInfo::getAddress).collect(toSet());
-            ClassLoader jobCl = jobClassloaderService.getClassLoader(jobId);
-            // We don't create the CL for light jobs.
-            assert jobClassloaderService.getClassLoader(jobId) == null;
-            doWithClassLoader(
-                    jobCl,
-                    () -> execCtx.initialize(coordinator, addresses, plan)
-            );
-        } catch (Throwable e) {
-            completeExecution(execCtx, new CancellationException());
-            throw e;
```

Comment:
Previously, we were rethrowing the exception here and this thrown exception is handled differently than the exception propagated through the returned future. See the exception handling in the `AsyncJobOperation` here: https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AsyncOperation.java#L56-L60. 
In this place, we're logging the thrown exceptions as an error but for the exceptions propagated through the future, we're not doing the same. I thought for handling this but I couldn't find a proper way for solving this discrepancy. I can only think of an ugly way that logging the exceptions in the more deeper place rather than inside the `AsyncOperation`. 
cc: @viliam-durina, @frant-hartm what do you suggest for handling this? 

By the way, this is my only remaining review item, other than that this PR looks good to me.

Reply:
Do you have any particular example?
The whenComplete() section also completes the execution context. whenComplete cannot recover from exceptions, so it's carried over to the caller - just like "throw e" is doing here, just that it's wrapped in ExecutionException. I _think_ the handling is more or less the same, maybe I'm missing something